### PR TITLE
Types

### PR DESCRIPTION
**Pull request by sbrunner**
## Description

Activate more mypy options, add the missing types

### CODE REVIEW

File: prospector/config/__init__.py [hunk 1]
```diff
@@ -168,7 +164,7 @@ def _get_profile(self, workdir: Path, config):
 
         # Use the strictness profile only if no profile has been given
         if config.strictness is not None and config.strictness:
-            cmdline_implicit.append("strictness_%s" % config.strictness)
+            cmdline_implicit.append("strictness_%s" % config.strictness.value)
```
Comment (Pierre-Sassoulas) [inline prospector/config/__init__.py:167]: This change the value (was it a bug ?)

File: prospector/config/configuration.py [hunk 2]
```diff
@@ -1,325 +1,316 @@
...
-        parser_options=parser_options,
-        positional=positional,
+    parser.add_argument(
+        "path",
+        help="The path to a Python project to inspect. Defaults to PWD if not specified. If multiple paths are specified, they must all be files (no directories).",
+        metavar="PATH",
+        nargs="?",
     )
+
+    return parser
```
Comment (Pierre-Sassoulas) [inline prospector/config/configuration.py:316]: The whole file is really hard to review, I'll trust you on this.

**Review by Pierre-Sassoulas — COMMENTED** [review]
I feel there's a lot of changes not directly related to the typing.

**Comment by sbrunner** [comment]
Effectively, this PR is relay not ready yet.

**Comment by sbrunner** [comment]
Replaced by  #687

**Pull request closed without merging**